src/types/agentTypes.ts

```diff
@@ -65,6 +66,20 @@ export const messageSchema = z.union([taskSchema, nonTaskScehma]);
 export type Task = z.infer<typeof taskSchema>;
 export type Message = z.infer<typeof messageSchema>;
 
+/* Agent Type */
+export const [AGENT_STATUS_RUNNING, AGENT_STATUS_STOPPED] = [
+  "running" as const,
+  "stopped" as const,
+];
+
+const AgentStatusSchema = z.union([
+  z.literal(AGENT_STATUS_RUNNING),
+  z.literal(AGENT_STATUS_STOPPED),
+  z.literal(""),
+]);
+
+export type AgentStatus = z.infer<typeof AgentStatusSchema>;
+
```

Comment:
This seems a hard to work with. Any reason we cant do the below?
```suggestion
export type AgentStatus = "running" | "stopped"
```

Reply:
mostly because the strings "running" and "stopped" is also used in many places in the code

Reply:
but on 2nd thought, I can make this more concise.